designdocs/permissions.md

```diff
@@ -0,0 +1,87 @@
+# proposal for permissions request
+
+
+### Philosophy
+The user-agent will always be closer to the user than a spec. The user agent has greater access to current conditions than we (the spec writers) ever will. As a philosophy we should give the user agent *as much freedom as possible to “do the right thing”* under whatever circumstances are at the time a web application is started.
+
+### scope
+
+Currently we are having difficulties finding a solution that all parties can agree to. This is about the permissions for various properties in WebXR.  This proposal addresses the following concerns
+
+* what permissions are available to be asked for
+* how granular are these permissions
+* are the permissions granted at page load / application start
+* or are the permissions granted at the time of use.
+
+### stating concerns
+
+Concerns about granting at page load time are
+* users may be requested to approve multiple permissions one after another, creating click fatigue.
+* if all apps request permissions at start, the user has no information to judge whether or not they should grant this, because they haven’t entered the experience yet.
+
+Concerns about granting requests once already inside of immersive mode are:
+* access requests when inside immersive mode should be done with some sort of a secure method, such as a dedicated hardware button, so that applications cannot spoof such permissions.
+* some devices or scenarios may not be able to provide such a secure method, thus doing the request before entering immersive mode is preferable.
+
```

Comment:
I'm still trying to read between the lines here and failing :-/

Are you saying that user agents MUST NOT request user consent both at the beginning and during a session?  Even if the consent being requested mid-session is unrelated to the consent granted at the beginning?

I think the root of my confusion is that I'm struggling to differentiate between you're saying Chrome is planning to do vs. what y'all are saying must be requirements on all user agents.

Reply:
(to be clear, I'm not taking a stance either way! I'm just trying to ensure I'm understanding your stance accurately)

Reply:
To clarify, my comments are general about what the spec (and this doc) say, which are related to general guidance and requirements that apply to all user agents.

I'm just raising the concern so that it can be considered and addressed. My comment was not intended to imply any requirements (i.e., "MUST NOT request..."). While that is one possible solution, there may be others.

FYI, I filed #702 for related issues.